openedx/openedx-platform
# Allow each instance to configure restrictions on allowed email addresses

Introduces a new optional setting, `REGISTRATION_EMAIL_PATTERNS_ALLOWED`, which can be used to restrict registration so that only users whose emails match a whitelist of allowed email patterns can register on the platform.

Use cases include private schools, corporate instances, etc. where employees/students are allowed to enroll themselves in courses but the general public should not be using the content.

This is something that a couple different OpenCraft clients have requested from time to time. We've recently implemented it for [Unow](http://www.unow.fr/), and we'd like to contribute this patch to the project.

Note that this does not completely prevent all use of the platform or enrollment in courses - it is still possible for people to enter a bogus email that matches against the whitelist and use the platform until required to activate their account.

JIRA: [OSPR-1023](https://openedx.atlassian.net/browse/OSPR-1023)

**Discussions**:
https://groups.google.com/forum/#!topic/edx-code/xK6bb1koToo

**Sandbox**:
LMS: http://pr10943.sandbox.opencraft.com/ 
Studio: http://studio.pr10943.sandbox.opencraft.com/
(Note: this works with both the "combined" and normal registration page but the sandbox only shows the normal one since that's our default sandbox config.)

**Test Instructions**:
1. Go to http://pr10943.sandbox.opencraft.com/register 
2. Try to register an account with an email like "blah@example.com". It should fail. Note: Please use "@example.com" or an invalid TLD like "@testing.none" because if you use a real domain name, someone may receive unexpected email from our sandbox.
3.  Try to register with (your) "@edx.org" or "@opencraft.com" email address. It should work (see whitelist in the "settings" below).
4. Go to http://studio.pr10943.sandbox.opencraft.com/signup and repeat steps 2-3. Use a different email address in step 2, since whatever email you used the first time will already be taken.  (something like "bounceme123@opencraft.com" should work).

---

**Settings**

``` yaml
EDXAPP_ENV_EXTRA:
  "REGISTRATION_EMAIL_PATTERNS_ALLOWED": [".*@edx\\.org", "^.*@opencraft\\.com$"]
```
